src/java.base/share/classes/java/util/zip/ZipOutputStream.java

```diff
@@ -144,11 +144,14 @@ public ZipOutputStream(OutputStream out, Charset charset) {
      *            ZIP file comment is greater than 0xFFFF bytes
      */
     public void setComment(String comment) {
+        byte[] bytes = null;
         if (comment != null) {
-            this.comment = zc.getBytes(comment);
-            if (this.comment.length > 0xffff)
-                throw new IllegalArgumentException("ZIP file comment too long.");
+            bytes = zc.getBytes(comment);
+            if (bytes.length > 0xffff) {
+                throw new IllegalArgumentException("ZIP file comment too long");
+            }
         }
+        this.comment = bytes;
```

Comment:
The implementation change looks okay. I assume this regression slipped through due to lack of tests.

The method description doesn't make it clear that the comment can be null (ZipEntry.setComment has the same issue) so we should fix this while we are in the area, as a separate JBS of course as it will need a CSR to track the spec clarification.

Reply:
ZipEntry::setComment indicates that the comment will be truncated if needed and ZipOutputStream takes care of this.

Perhaps writeEND() should also be updated to  something like:
`writeBytes(comment, 0, Math.min(comment.length, 0xffff))`

Which is similar to what happens in writeCEN

Yes it would be nice to clarify that a null is accepted by setComment.  When null is specified, the comment length is written as 0

Reply:
> Yes it would be nice to clarify that a null is accepted by setComment. When null is specified, the comment length is written as 0

@mrserb Are you taking the spec clarification or should we just created another issue in JBS to track it?

Reply:
I would like to update the spec for the null value for this method, and probably others in a separate CR, since this fix could be backported to the early releases. Will create such CR after agreement.

Reply:
I have filed a separate issue: https://bugs.openjdk.java.net/browse/JDK-8277495